presets/ragengine/inference/inference.py

```diff
@@ -38,29 +51,99 @@ def complete(self, prompt: str, **kwargs) -> CompletionResponse:
             self.params = {}
 
     def _openai_complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
-        llm = OpenAI(
-            api_key=LLM_ACCESS_SECRET,
-            **kwargs  # Pass all kwargs directly; kwargs may include model, temperature, max_tokens, etc.
-        )
-        return llm.complete(prompt)
+        return OpenAI(api_key=LLM_ACCESS_SECRET, **kwargs).complete(prompt)
 
     def _huggingface_remote_complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
-        headers = {"Authorization": f"Bearer {LLM_ACCESS_SECRET}"}
-        data = {"messages": [{"role": "user", "content": prompt}]}
-        response = requests.post(LLM_INFERENCE_URL, json=data, headers=headers)
-        response_data = response.json()
-        return CompletionResponse(text=str(response_data))
+        return self._post_request(
+            {"messages": [{"role": "user", "content": prompt}]},
+            headers={"Authorization": f"Bearer {LLM_ACCESS_SECRET}"}
+        )
 
     def _custom_api_complete(self, prompt: str, **kwargs: Any) -> CompletionResponse:
-        headers = {"Authorization": f"Bearer {LLM_ACCESS_SECRET}"}
+        model = kwargs.pop("model", self._get_default_model())
         data = {"prompt": prompt, **kwargs}
+        if model:
+            data["model"] = model # Include the model only if it is not None
+
+        # DEBUG: Call the debugging function
+        # self._debug_curl_command(data)
+        try:
+            return self._post_request(data, headers=DEFAULT_HEADERS)
+        except Exception as e:
+            err_msg = str(e)
+            # Check for vLLM-specific missing model error
+            if "missing" in err_msg and "model" in err_msg and "Field required" in err_msg:
+                logger.warning(
+                    f"Detected missing 'model' parameter in API response. "
+                    f"Response: {err_msg}. Attempting to fetch the default model..."
+                )
+                self._default_model = self._fetch_default_model()  # Fetch default model dynamically
+                if self._default_model:
+                    logger.info(f"Default model '{self._default_model}' fetched successfully. Retrying request...")
+                    data["model"] = self._default_model
+                    return self._post_request(data, headers=DEFAULT_HEADERS)
+                else:
+                    logger.error("Failed to fetch a default model. Aborting retry.")
+            raise # Re-raise the exception if not recoverable
```

Comment:
Note here we hit inference endpoint. If we are returned with a missing `model` parameter error then we attempt to hit models endpoint. 

This makes the check targeted, as it will retry hitting this endpoint as long as there is a missing models parameter error.